project/Build.scala

```diff
@@ -915,34 +917,47 @@ object Build {
       unmanagedSourceDirectories in Test +=
         baseDirectory.value.getParentFile.getParentFile / "shared/src/test/scala",
 
-      if (isGeneratingForIDE) {
-        unmanagedSourceDirectories in Test +=
-          baseDirectory.value.getParentFile.getParentFile / "shared/src/test/scala-ide-stubs"
-      } else {
-        sourceGenerators in Test += Def.task {
+      unmanagedSourceDirectories in Test ++= {
+        if (isGeneratingForIDE)
+          Seq(baseDirectory.value.getParentFile.getParentFile / "shared/src/test/scala-ide-stubs")
+        else
+          Seq()
+      },
+
+      BuildInfoPlugin.buildInfoScopedSettings(Test),
+      BuildInfoPlugin.buildInfoDefaultSettings,
+      buildInfoPackage in Test := "org.scalajs.linker.testutils",
+      buildInfoObject in Test := "StdlibHolder",
+      buildInfoOptions in Test += BuildInfoOption.PackagePrivate,
+      buildInfoKeys in Test := {
+        val previousLibsTask = Def.task {
```

Comment:
Perhaps this deserves a separate, named task, now?

Reply:
I have attempted this, but we'd either have to:

- Not type it properly (`Map[String, String]` instead of `Map[String, File]`).
- Re-shape it again in the `buildInfoKeys` settings (map `File` to `String` via `getAbsolutePath`).

I don't think with this in mind it is worth it. WDYT?

Reply:
Hum yes, I see. Let's leave it as is, then.